src/app/util/sqidStatementTable.directive.js

```diff
@@ -223,7 +233,22 @@ function($compile, Properties, dataFormatter, util, i18n) {
 		scope.getShowRowsClass = function(id) {
 			return (scope.showRows(id) ? 'expand-open' : 'expand-closed' );
 		}
-		
+		scope.approve = function(statement){
+			primarySources.approve(scope.id, decodeURIComponent(statement), true);
```

Comment:
Suggestion for a more elegant approach that avoids pulling primarySources into this module: You could set the approve and reject functions as values of the statement or reference to be approved. The signature of these functions would always be the same, so the calls here don't need to know they came from primarySources.

This approach has some advantages:
* The code that inserts the proposals can define the approve/reject functions directly
* This module here does not have to know about all possible proposal sources, and one would not need to introduce a switch-like statement to select the correct module from the source in the future
* By not setting a function for reject, you can signal that a proposal cannot be rejected.
* The approach allows you to add further optional functions in the future (approve, reject, explain [find out why this was proposed], delete [if a proposal is to delet an existing statement that seems wrong], ...).

One would probably need to access the objects that should be approved/rejected through the $scope instead of encoding them into URLs to make this work (as remarked in another comment). Maybe this is easy without much data duplication, since the $scope has the statements already, so one just needs to refer to them in some stable way. If this fails, one can still add the statements/references to an array in $scope and refer to their index in the function call.